community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -876,19 +874,21 @@ public synchronized void stop()
             awaitAllTransactionsClosed();
 
             // Write new checkpoint in the log only if the kernel is healthy.
-            // We cannot throw here since we need to shutdown without exceptions.
-            if ( databaseHealth.isHealthy() )
+            // We cannot throw here since we need to shutdown without exceptions,
+            // so let's make the checkpointing part of the life, so LifeSupport can handle exceptions properly
+            life.add( new LifecycleAdapter()
```

Comment:
Perhaps this is a neat way of solving it, I just want this checkpointer lifecycle instance to be added during startup, not during calling stop().